examples/ping/src/main.rs

```diff
@@ -21,31 +21,21 @@
 #![doc = include_str!("../README.md")]
 
 use futures::prelude::*;
-use libp2p::core::upgrade::Version;
-use libp2p::{
-    identity, noise, ping,
-    swarm::{SwarmBuilder, SwarmEvent},
-    tcp, yamux, Multiaddr, PeerId, Transport,
-};
+use libp2p::{noise, ping, swarm::SwarmEvent, tcp, yamux, Multiaddr};
 use std::error::Error;
-use std::time::Duration;
 
 #[async_std::main]
 async fn main() -> Result<(), Box<dyn Error>> {
-    env_logger::init();
-    let local_key = identity::Keypair::generate_ed25519();
-    let local_peer_id = PeerId::from(local_key.public());
-
-    let transport = tcp::async_io::Transport::default()
-        .upgrade(Version::V1Lazy)
-        .authenticate(noise::Config::new(&local_key)?)
-        .multiplex(yamux::Config::default())
-        .boxed();
-
-    let mut swarm =
-        SwarmBuilder::with_async_std_executor(transport, ping::Behaviour::default(), local_peer_id)
-            .idle_connection_timeout(Duration::from_secs(60)) // For illustrative purposes, keep idle connections alive for a minute so we can observe a few pings.
```

Comment:
Missing idle connection timeout in the new version.

Reply:
99d7b6dc2